src/Components/FormCreation/CardComponents/FileQuestion.tsx

```diff
@@ -16,37 +18,55 @@ type Props = {
   submission: boolean;
   short_answer: boolean;
   validation?: any;
-  onChange: (options: any) => void;
+  onChange: (options: Array<string> | number) => void;
   active: boolean;
   initialNumFiles?: number;
 };
 
+const useStyles = makeStyles({
+  button: {
+    height: 36,
+    marginTop: 16,
+    marginBottom: 24,
+    marginLeft: 16
+  }
+});
+
 //TODO: Add Response Validation
 function FileQuestion({
   submission,
   active,
-  validation,
   onChange,
   initialNumFiles
-}: Props) {
+}: Props): React.ReactElement {
   const [numFiles, setnumFiles] = useState(initialNumFiles || 1);
-  const handleNumFileChange = (value: any) => {
+  const classes = useStyles();
+  const [spinner, setSpinner] = useState(false);
+  const handleNumFileChange = (value: number | null | unknown) => {
     //Check validations here and update the error prop accordingly
     setnumFiles(value as number);
-    onChange(numFiles);
+    onChange(value as number);
   };
+  const [files, setFiles] = useState([] as Array<string>);
 
   const numFileOptions = [1, 2, 3, 5, 10];
 
-  async function handleFile(event: any) {
-    const file = event.target.files[0]; //access the file
-    //todo: check if the file extension is valid
-    const formData = new FormData();
-    formData.append("file", file); // appending file
+  async function handleFileUpload(filesToUpload: FileList | null) {
+    if (!filesToUpload) return;
+    setSpinner(true);
+    const file = filesToUpload[0]; //access the file
 
-    //returns the url to the file in aws
-    await fileUpload("arbitrium", file.name, formData);
-    //TODO Save to DB
+    if (file) {
+      //todo: check if the file extension is valid
+      const formData = new FormData();
+      formData.append("file", file); // appending file
+
+      //returns the url to the file in aws
+      const result = await fileUpload("arbitrium", file.name, formData);
```

Comment:
Might be a dumb question but in a previous file I saw "arbitrium-public". I assume conversely we have "arbitrium-private"? Should this line be using one of those two?

Reply:
"aribtrium" is the private one.